bull/pdl-queue/consumer.js

```diff
@@ -121,42 +161,78 @@ async function uploadToIA(zip, metadata, byteLength, email, job) {
-  jobLogs["userName"] = job.data.details.userName;
-  const [zip, byteLength, errorFlag] = await getZipAndBytelength(
-    job.data.details.Pages,
-    job.data.details.bookID,
-    job.data.details.title,
-    job
+async function uploadPdfToIA(pdfBuffer, metadata, byteLength, email, job) {
+  const bucketTitle = metadata.IAIdentifier;
+  const IAuri = `http://s3.us.archive.org/${bucketTitle}/${bucketTitle}.pdf`;
+  let headers = setHeaders(
+    metadata,
+    byteLength,
+    metadata.title,
+    job.data.details.contentType
   );
-  if (errorFlag.status) {
-    job.log(JSON.stringify(jobLogs));
-    logUserData(jobLogs["userName"], "Panjab Digital Library");
-    logger.log({
-      level: "error",
-      message: `Upload to Internet Archive failed because ${errorFlag.page} is not reachable. Please try again or contact Panjab Digital Library for more details.`,
-    });
-    job.progress(100);
-    done(
-      new Error(
-        `Upload to Internet Archive failed because <a href=${errorFlag.page} target='_blank'>${errorFlag.page}</a>  is not reachable. Please try again or contact Panjab Digital Library for more details.`
-      )
-    );
-  } else {
+  const options = {
+    method: "PUT",
+    uri: IAuri,
+    headers: headers,
+  };
+  const readableStream = Readable.from(pdfBuffer);
+  readableStream.pipe(
+    request(options, async (error, response, body) => {
+      if (response.statusCode === 200) {
+        // EmailProducer(email, metadata.title, IAuri, true);
+      } else {
+        logger.log({
+          level: "error",
+          message: `IA Failure PDL ${body}`,
+        });
+      }
+    })
+  );
+}
+
+PDLQueue.process(async (job, done) => {
+  try {
+    const jobLogs = job.data.details;
+    const trueURI = `http://archive.org/details/${job.data.details.IAIdentifier}`;
+    jobLogs["trueURI"] = trueURI;
+    jobLogs["userName"] = job.data.details.userName;
     job.log(JSON.stringify(jobLogs));
     logUserData(jobLogs["userName"], "Panjab Digital Library");
-    job.progress(90);
-    await uploadToIA(
-      zip,
-      job.data.details,
-      byteLength,
-      job.data.details.email,
-      job
-    );
-    job.progress(100);
-    done(null, true);
+
+    if (job.data.details.pdfUrl) {
+      const { pdfBuffer, byteLength } = await getPdfAndBytelength(
+        job.data.details.pdfUrl,
+        job
+      );
+      await uploadPdfToIA(
+        pdfBuffer,
+        job.data.details,
+        byteLength,
+        job.data.details.email,
+        job
+      );
+      job.progress(100);
+      done(null, true);
+    } else {
+      const [zip, byteLength] = await getZipAndBytelength(
+        job.data.details.Pages,
+        job.data.details.bookID,
+        job.data.details.title,
+        job
+      );
+      job.progress(90);
+      await uploadZipToIA(
+        zip,
+        job.data.details,
+        byteLength,
+        job.data.details.email,
+        job
+      );
+      job.progress(100);
+      done(null, true);
```

Comment:
What is the job fails? It will still return success.

Reply:
@coderwassananmol  I have pushed a fix for this